File: platform/polyglot/github_sdk_api_component_usage_audit/mjs/src/github/endpoints/repo-meta.mjs
/**
 * Repository Metadata Endpoint
 *
 * Fetches /repos/{owner}/{repo} with an in-memory cache keyed by "owner/repo".
 * Validates stargazers_count and archived status.
 */

/**
 * Fetch and validate repository metadata.
 *
 * @param {object} ctx - Shared context from the service.
 * @param {Function} ctx.makeRequest - SDK core request function.
 * @param {Function} ctx.debugLog - Debug logger function.
 * @param {Map} ctx.cache - In-memory cache map.
 * @param {object} params
 * @param {string} params.owner - Repository owner.
 * @param {string} params.repo - Repository name.
 * @param {number} params.minStars - Minimum stargazer count.
 * @returns {Promise<{ valid: boolean, repo?: object }>}
 */
export async function fetchRepoMeta(ctx, { owner, repo, minStars }) {
  const cacheKey = `${owner}/${repo}`;

  // Check cache
  if (ctx.cache.has(cacheKey)) {
    return ctx.cache.get(cacheKey);
  }

  const data = await ctx.makeRequest(`GET /repos/${owner}/${repo}`);

  await ctx.debugLog("repo-meta", { owner, repo, stars: data.stargazers_count, archived: data.archived });

  const valid = data.stargazers_count >= minStars && data.archived === false;

  const result = { valid, repo: data };
  ctx.cache.set(cacheKey, result);

  return result;
}

File: platform/polyglot/github_sdk_api_component_usage_audit/py/github_sdk_api_component_usage_audit/sdk/repo_meta.py
"""
Repository Metadata Endpoint

Fetches /repos/{owner}/{repo} with an in-memory cache keyed by "owner/repo".
Validates stargazers_count and archived status.
"""

from __future__ import annotations

import logging
from typing import Any

from github_api.sdk.client import GitHubClient

__all__ = ["fetch_repo_meta"]

logger = logging.getLogger("github_sdk_api_component_usage_audit.sdk.repo_meta")


async def fetch_repo_meta(
    client: GitHubClient,
    *,
    owner: str,
    repo: str,
    min_stars: int,
    cache: dict[str, dict[str, Any]],
) -> dict[str, Any]:
    """Fetch and validate repository metadata.

    Args:
        client: Authenticated GitHubClient instance.
        owner: Repository owner.
        repo: Repository name.
        min_stars: Minimum stargazer count.
        cache: In-memory cache dict keyed by "owner/repo".

    Returns:
        Dict with ``valid`` bool and optional ``repo`` data.
    """
    cache_key = f"{owner}/{repo}"

    if cache_key in cache:
        return cache[cache_key]

    data = await client.get(f"/repos/{owner}/{repo}")

    logger.debug(
        "repo-meta: %s/%s stars=%d archived=%s",
        owner,
        repo,
        data.get("stargazers_count", 0),
        data.get("archived", False),
    )

    valid = (
        data.get("stargazers_count", 0) >= min_stars
        and data.get("archived") is False
    )

    result = {"valid": valid, "repo": data}
    cache[cache_key] = result

    return result
